src/commands/deployWorkspaceProject/deployWorkspaceProject.ts

```diff
@@ -30,33 +33,45 @@ import { ShouldSaveDeploySettingsPromptStep } from "./ShouldSaveDeploySettingsPr
 import { DefaultResourcesNameStep } from "./getDefaultValues/DefaultResourcesNameStep";
 import { getDefaultContextValues } from "./getDefaultValues/getDefaultContextValues";
 
-export async function deployWorkspaceProject(context: IActionContext, item?: ContainerAppItem | ManagedEnvironmentItem): Promise<void> {
-    ext.outputChannel.appendLog(localize('beginCommandExecution', '--------Initializing deploy workspace project--------'));
+export async function deployWorkspaceProject(context: IActionContext & Partial<DeployWorkspaceProjectContext>, item?: ContainerAppItem | ManagedEnvironmentItem): Promise<DeployWorkspaceProjectResults> {
+    ext.outputChannel.appendLog(
+        context.invokedFromApi ?
+            localize('beginCommandExecutionApi', '--------Initializing deploy workspace project (Azure Container Apps - API)--------') :
+            localize('beginCommandExecution', '--------Initializing deploy workspace project--------'));
 
     // If an incompatible tree item is passed, treat it as if no item was passed
     if (item && !ContainerAppItem.isContainerAppItem(item) && !ManagedEnvironmentItem.isManagedEnvironmentItem(item)) {
         item = undefined;
     }
 
-    const subscription: AzureSubscription = await subscriptionExperience(context, ext.rgApiV2.resources.azureResourceTreeDataProvider);
+    const subscription: AzureSubscription = context.subscription ?? await subscriptionExperience(context, ext.rgApiV2.resources.azureResourceTreeDataProvider);
     const subscriptionContext: ISubscriptionContext = createSubscriptionContext(subscription);
 
     // Show loading indicator while we configure default values
     let defaultContextValues: Partial<DeployWorkspaceProjectContext> | undefined;
     await window.withProgress({
         location: ProgressLocation.Notification,
         cancellable: false,
-        title: localize('loadingWorkspaceTitle', 'Loading workspace project deployment configurations...')
+        title: context.invokedFromApi ?
+            undefined :
+            localize('loadingWorkspaceTitle', 'Loading workspace project deployment configurations...')
     }, async () => {
         defaultContextValues = await getDefaultContextValues({ ...context, ...subscriptionContext }, item);
     });
 
+    let activityContext: Partial<ExecuteActivityContext>;
```

Comment:
Don't show activity log updates in ACA when another client function calls into this API.  Let the client handle showing its own activity log updates